src/main/java/com/cloudbees/jenkins/support/filter/FilteredInputStream.java

```diff
@@ -0,0 +1,65 @@
+package com.cloudbees.jenkins.support.filter;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.function.Function;
+
+/**
+ * A custom {@link InputStream} that filters lines of text read from an underlying {@link InputStream} using a provided
+ * mapping function. Each line read from the input stream is processed by the function before being returned, allowing
+ * for modifications such as redaction or transformation of the line content.
+ *
+ * @author Basil Crow
+ */
+public class FilteredInputStream extends InputStream {
```

Comment:
Could possibly be replaced by https://javadoc.jenkins.io/hudson/console/LineTransformationOutputStream.Delegating.html

Reply:
Is that not an `OutputStream`, while in this case we need an `InputStream`?